derive/src/pymodule.rs

```diff
@@ -37,6 +37,36 @@ pub fn impl_pymodule(attr: AttributeArgs, module_item: Item) -> Result<TokenStre
     // collect to context
     for item in items.iter_mut() {
         let r = item.try_split_attr_mut(|attrs, item| {
+            // If attribute is #[pyattr] and item is ItemFn, then
```

Comment:
this is causing issues someplace but I can't track it down. Removing it from here and moving the static_cell to `new_exception_type` should be a viable alternative though.

Reply:
Moving `static_cell` to `new_exception_type` may solve that test failures. But we need to declare it as macro as each functions need its own `static_cell`. Approach with procedural-macro is best I think as it also solve other stuffs (#[pyattr] attached function in module which is not exception-type)

After more trying, I would mention you. sorry 😢

Reply:
Ah, right. Adding an `error` argument to `pyattr` then? Have it only applied on attributes that are meant to denote errors.

Reply:
Oh . I missed your comment. I add "once" keyword to #[pyattr]  like #[pyattr(name = "blahblah", once)] for that exceptions.

Reply:
I am not sure this is fit for here, but adding a field to `ALLOWED_NAMES` allows to read it from `ItemMeta`.
(with a new `AttrItemMeta`)
If you are interested in, check how `ClassItemMeta` handles `metaclass`. If this is not fit for here, still `AttrItemMeta` is preferred not to allow other simple items to have `once`.

Reply:
Sorry for begin late 😂 I moved static_cell wrapping code to `get_module_item` of `AttributeItem`.
Use `AttrItemMeta::from_attr` for `ItemFn` and replace original once keyword checking code to `attr_meta.inner()._bool("once")?` . For the other cases, `Item::Const` and `Item::Use`, just use `SimpleItemMeta::from_attr` as they do not have `once` keyword ever. Please take a look this commit [7f2fd97](https://github.com/RustPython/RustPython/pull/3402/commits/7f2fd97278905bf9f3e7e8bd834b89bfb7fda1e8). Thanks 😊